### PR TITLE
Optimize CI a bit

### PR DESCRIPTION
**Pull request by Goooler**
- Run CI on main branch.
- Bump actions.
- Use `gradle/gradle-build-action` to speed up builds.

https://github.com/actions/checkout/releases/tag/v3.0.0 
https://github.com/actions/setup-java/releases/tag/v3.0.0 
https://github.com/gradle/gradle-build-action

### CODE REVIEW

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -1,16 +1,46 @@
 name: Android CI
-on: [pull_request]
 
-jobs:
-  build:
+on:
+  push:
+    branches:
+      - main
```
Comment (cketti) [inline .github/workflows/android.yml:6]: I don't think we need a CI run when `main` is updated. All changes are made via pull requests. Lately I also rebase them onto `main`. But even without that, it's very unlikely there's a problem where tests pass on the PR branch, but fail when the branch is merged into main. At least I can't remember this happening in the past. If it does happen, the next pull request will catch that case.

File: .github/workflows/android.yml [hunk 2]
```diff
@@ -1,16 +1,46 @@
...
+      - '**.md'
+      - 'LICENSE'
+      - 'NOTICE'
 
+jobs:
+  validation:
+    name: "Validation"
     runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
```
[the remark this answers is not in the document]
Comment (Goooler) [reply .github/workflows/android.yml]: Done.

File: .github/workflows/android.yml (hunk 1 above)
Comment (Goooler) [reply .github/workflows/android.yml]: Reverted.

Comment (cketti) [reply .github/workflows/android.yml]: I did like the `paths-ignore` lines.

Comment (Goooler) [reply .github/workflows/android.yml]: Done.

**Pull request merged**